### DISCUSSION TITLE
wagmi writeContract with walletconnect

### DISCUSSION THREAD

**Discussion by roycebank** [WalletConnect/walletconnect-monorepo#2048] [Developer Support]
hi i am trying to write to a contract how can i do this in true way? i am keep getting error messages. 
i am using these packages

`import { configureChains, createClient, sendTransaction, prepareSendTransaction, prepareWriteContract, writeContract } from "@wagmi/core";
import { bsc } from "@wagmi/core/chains";
import { Web3Modal } from "@web3modal/html";
import {
  EthereumClient,
  modalConnectors,
  walletConnectProvider,
} from "@web3modal/ethereum";`

i tried this:
`const { config } = await prepareWriteContract({
    address: pancakeSwapRouterAdress,
    abi: JSON.parse(pancakeAbi),
    functionName: 'swapExactETHForTokens',
    args: [web3.utils.toHex("0"), [wbnb, tokenToBuy], connectedW,  web3.utils.toHex( (Math.round(Date.now() / 1000) + 60 * 20).toString() ) ],
    overrides: {
      from : connectedW,
      value: web3.utils.toHex("10000000"),
    },
    chainId : bsc.id
  });

  const { write } = await writeContract(config)`

and failes with "TypeError: Cannot read properties of undefined (reading 'chainId')"

i also tried with sendTransaction, prepareSendTransaction, gave data to prepareSendTransaction, i saw the transaction screen on my metamask but as unknown method. any help is appreciated

**Comment by webnatrix** [comment]
The same problem I was able to read the contract, but now when I write to getContract I pass the user and it does not perform those actions. I can't understand why.
How did you solve the problem?

**Reply by webnatrix** [reply]
```
const contract = getContract({
  address: "{{ $item->getNetworkTokenID->address }}", 
  abi: {!! $item->getNetworkTokenID->getNetwork->abiERC20 !!},
  walletClient: getWalletClient
});

var approve = await contract.write.approve([pool, 0])
```

I tried this I don't know how to getWalletClient() it returns NULL

**Reply by webnatrix** [reply]
https://github.com/wagmi-dev/wagmi/blob/c8139b1cd5b9cc78961e158cc6f0de59f198908f/packages/core/src/actions/contracts/getContract.test-d.ts#L23

It shows that if you pass it, you get a write

**Reply by roycebank** [reply]
bro if you want to example code i can share it but WalletConnect v1.0 has been shutted down 4 days ago

**Reply by webnatrix** [reply]
I had everything super on version 2, but now they have attached Veim and everything is broken (

**Reply by roycebank** [reply]
:(

**Reply by webnatrix** [reply]
Did you manage to recreate the situation?

**Reply by webnatrix** [reply]
New WC2 https://github.com/webnatrix/web3modal2test
Error tx ( not work
